File: uniffi/src/message/groups/trans_last_idx_sig_groups.rs
use cesrox::error::ParsideResult;
use cesrox::Group;
use cesrox::Prefixer;
pub use cesrox::message::groups::{
    TransLastIdxSigGroups,
    TransLastIdxSigGroup,
    ControllerIdxSigs
};

pub fn trans_last_idx_sig_group_create(prefixer: Prefixer, isigers: ControllerIdxSigs) -> TransLastIdxSigGroup {
    TransLastIdxSigGroup::new(prefixer, isigers)
}

pub fn trans_last_idx_sig_groups_create(value: Vec<TransLastIdxSigGroup>) -> TransLastIdxSigGroups {
    TransLastIdxSigGroups::new(value)
}

pub fn trans_last_idx_sig_groups_qb64(trans_last_idx_sig_groups: &TransLastIdxSigGroups) -> ParsideResult<String> {
    trans_last_idx_sig_groups.qb64()
}

pub fn trans_last_idx_sig_groups_qb64b(trans_last_idx_sig_groups: &TransLastIdxSigGroups) -> ParsideResult<Vec<u8>> {
    trans_last_idx_sig_groups.qb64b()
}

pub fn trans_last_idx_sig_groups_qb2(trans_last_idx_sig_groups: &TransLastIdxSigGroups) -> ParsideResult<Vec<u8>> {
    trans_last_idx_sig_groups.qb2()
}

File: wrappers/kotlin/lib/src/main/kotlin/cesrox/groups/TransLastIdxSigGroups.kt
package cesrox.groups

import cesrox.primitives.Prefixer
import uniffi.cesrox.*
import uniffi.cesrox.TransLastIdxSigGroup as TransLastIdxSigGroupNative
import uniffi.cesrox.TransLastIdxSigGroups as TransLastIdxSigGroupsNative

class TransLastIdxSigGroup {
    val value: TransLastIdxSigGroupNative

    constructor(prefixer: Prefixer, isigers: ControllerIdxSigs) {
        value = transLastIdxSigGroupCreate(prefixer.value, isigers.value)
    }

    constructor(nativeValue: TransLastIdxSigGroupNative) {
        value = nativeValue
    }

    val prefixer get() = Prefixer(value.prefixer)

    val isigers get() = ControllerIdxSigs(value.isigers)
}

class TransLastIdxSigGroups {
    val value: TransLastIdxSigGroupsNative

    constructor(couples: List<TransLastIdxSigGroup>) {
        value = transLastIdxSigGroupsCreate(couples.map { it.value })
    }

    constructor(couples: TransLastIdxSigGroupsNative) {
        value = couples
    }

    fun get(index: Int): TransLastIdxSigGroup {
        return TransLastIdxSigGroup(value.value[index])
    }

    fun qb2(): List<UByte> {
        return transLastIdxSigGroupsQb2(value)
    }
    fun qb64b(): List<UByte> {
        return transLastIdxSigGroupsQb64b(value)
    }
    fun qb64(): String {
        return transLastIdxSigGroupsQb64(value)
    }
}